report/renderer/api.js

```diff
@@ -71,10 +75,22 @@ function createStylesElement() {
   return dom.createComponent('styles');
 }
 
+/**
+ * @param {HTMLElement} rootEl
+ * @param {Parameters<ReportUIFeatures['addButton']>[0]} opts
+ */
+function addButton(rootEl, opts) {
+  const features = rootElToReportUIFeatures.get(rootEl);
```

Comment:
We could get away with creating a new instance of `ReportUIFeatures` in this function since `ReportUIFeatures.addButton` only depends on the `_dom` member of `ReportUIFeatures`.

Reply:
This does work, however you have to dig through multiple class constructor to confirm thhere aren't any side effects from creates a new ReportUIFeatures, which suggests that this could be a footgun if those constructors began doing something other than rebinding a bunch of methods ... makes me lean towards keeping the map around. Semantically it makes since that there should only ever be one instance. wdyt?